docs: Uninstall Gateway

jamilbk opened the pull request:
https://discord.com/channels/1228082899023298741/1239053156965679115

jamilbk committed:
Checkpoint Gateway uninstall

jamilbk committed:
Add uninstall docs for Gateway

jamilbk commented on website/src/app/kb/administer/uninstall/readme.mdx:
```diff
@@ -0,0 +1,77 @@
+
+This guide covers how to uninstall the Firezone Gateway from your server(s).
+
+{/* add Alert and link to client uninstall docs when that's complete */}
```
```suggestion
{/* TODO: add Alert and link to client uninstall docs when that's complete */}
```

jamilbk committed:
Update website/src/app/kb/administer/uninstall/readme.mdx

Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
docs: Fix docs code blocks with arrow (#4982)

before

<img width="222" alt="Screenshot 2024-05-14 at 6 29 52 AM"
src="https://github.com/firezone/firezone/assets/167144/a9308eca-b6ae-417c-a673-94c7ee8cdec6">

after

<img width="209" alt="Screenshot 2024-05-14 at 8 31 40 AM"
src="https://github.com/firezone/firezone/assets/167144/1de92b63-8679-416a-9014-bf3593d141c4">

jamilbk committed:
Merge branch 'main' into docs/add-uninstall

jamilbk committed:
chore: bump versions for 1.0.4 release (#4985)

Update download links etc for the 1.0.4 release

AndrewDryga committed:
chore(portal): Use redactor to reduce chances of accidentally logging secrets (#4983)

bmanifold committed:
refactor(portal): Update IDP creation flow (#4984)

Why:

* The new flow for creating an identity provider in Firezone allows the
user to not have to worry what features their plan has enabled. It will
allow the user to select which identity provider they use and will take
them to the appropriate form depending on the features they have enabled
on their plan.

## Screenshots

### Selecting an identity provider
<img width="937" alt="Screenshot 2024-05-14 at 11 53 17 AM"
src="https://github.com/firezone/firezone/assets/2646332/31337ad9-13c8-43a2-942c-adb0a951167c">


### New OIDC form when a custom provider is selected but IDP sync is not
enabled for account
<img width="903" alt="Screenshot 2024-05-14 at 11 54 58 AM"
src="https://github.com/firezone/firezone/assets/2646332/2e18d788-60c3-4fad-b749-351559a24aca">

AndrewDryga committed:
Bump all Elixir deps

AndrewDryga committed:
Move GCP deployments to separate GitHub environments

ReactorScram committed:
chore(windows): style the MSI installer (#4990)

Closes #4919 

![image](https://github.com/firezone/firezone/assets/13400041/455bd444-3b75-4bfb-8cc0-afdcd29c55de)

![image](https://github.com/firezone/firezone/assets/13400041/b166fb9a-d7bc-4ac8-8ade-99cebfc7b7a5)

![image](https://github.com/firezone/firezone/assets/13400041/27eff651-08a4-4185-93cc-e1e544595ad8)

bmanifold commented on website/src/app/kb/administer/uninstall/readme.mdx:
````diff
@@ -0,0 +1,77 @@
+   docker stop firezone-gateway
+   ```
+
+1. Remove the container its volumes:
````
```suggestion
1. Remove the container and its volumes:
```

bmanifold commented on website/src/app/kb/administer/uninstall/readme.mdx:
```diff
@@ -0,0 +1,77 @@
+
+If you want to remove the Gateway's data from the admin portal as well, navigate
+to the Site where the Gateway was running, click `(see all)` next
+`Online Gateways`, and then delete the offline Gateway corresponding to the
```
```suggestion
to the Site where the Gateway was running, click `(see all)` next to the
`Online Gateways` header, and then delete the offline Gateway corresponding to the
```

bmanifold reviewed (approved):
Couple of tiny grammar things, but other than that LGTM

jamilbk committed:
Update readme.mdx

Co-authored-by: Brian Manifold <bmanifold@users.noreply.github.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
Update readme.mdx

Co-authored-by: Brian Manifold <bmanifold@users.noreply.github.com>
Signed-off-by: Jamil <jamilbk@users.noreply.github.com>

jamilbk committed:
docs: Refactor SSO guides to make sync optional (#4988)

Makes the sync steps optional so the Google, Okta, and Entra guides work
for all plans.

refs #4984

conectado committed:
feat(connlib): report resource status to client (#4931)

This PR introduces site's `Status`. That's used to report to the client
the status, either, unknown, online or offline, mostly as a hint to
users as what's wrong with a connection.

This are the criteria for an online or offline resource

* If all sites related to a resource are offline the resource is
considered offline, since there's no gateway that can respond to that
resource's connection
* If any site is online the resource is online, since that same peer can
be used to reach that resource
* Any other case is unknown

Right now resources are single site so it doesn't matter too much but
tracking online/offline per-site instead of per-gateway or resource
seems like the better long-term solution.

The way to "find out" the site's status is:

* If a response to a connection details is offline, all sites related to
that resource must be offline otherwise there would've been a gateway in
the response
* At the point we connect to a gateway, the site that corresponds to
that gateway must be online
* When a connection to a peer stops it's considered unknown again

Fixes #4738

jamilbk committed:
Merge branch 'main' into docs/add-uninstall
